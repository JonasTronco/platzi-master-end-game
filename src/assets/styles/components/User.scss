@import '../main.scss';

.user {
  @include flexStaCen;
}
.user__img {
  width: 75px;
  height: 75px;
  padding: 0;
  margin-bottom: $space_x2;
  margin-right: $space_x6;
  border-radius: 50%;
  overflow: hidden;
}
