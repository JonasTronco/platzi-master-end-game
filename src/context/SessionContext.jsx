import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
