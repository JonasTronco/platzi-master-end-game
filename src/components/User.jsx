import PropTypes from 'prop-types';
import '../assets/styles/components/User.scss';

function User({ user }) {
  return (
    <div className="user">
      <img
        className={`user__img ${user.isActive ? '' : 'disabled'}`}
        src={user.imageURL || 'https://i.imgur.com/oMJFiLX.jpg'}
        alt="user"
      />
      <div className="user__data">
        <h2 className="user__data__name">{`${user.firstName} ${user.lastName}`}</h2>
        <p className="user__data__role">{user.typeOfUser}</p>
      </div>
    </div>
  );
}

User.propTypes = {
  user: PropTypes.object.isRequired,
};

export default User;
