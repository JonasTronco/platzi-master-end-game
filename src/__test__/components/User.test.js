import User from '../../components/User';
import { create } from 'react-test-renderer';
import { SessionProvider } from '../../context/SessionContext';
describe('User Component', () => {
  test('Should render User component', () => {
    const component = create(
      <SessionProvider>
        <User user={{}} isActive={true} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
