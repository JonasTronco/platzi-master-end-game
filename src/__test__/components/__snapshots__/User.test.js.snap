// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Component Should render User component 1`] = `
<div
  className="user"
>
  <img
    alt="user"
    className="user__img--disabled"
    src="https://i.imgur.com/oMJFiLX.jpg"
  />
  <div
    className="user__data"
  >
    <h2
      className="user__data__name"
    >
      undefined undefined
    </h2>
    <strong
      className="user__data__role"
    />
    <p
      className="user__data__role"
    />
  </div>
</div>
`;
